### PR TITLE
Add opcode OP_HASH256

### PR DESCRIPTION
**Pull request by KhairallahA**
Closes: #157

### CODE REVIEW

File: src/script/engine.zig [hunk 1]
```diff
@@ -486,6 +487,20 @@ pub const Engine = struct {
...
+    /// OP_HASH256: The input is hashed two times with SHA-256.
+    fn opHash256(self: *Engine) EngineError!void {
+        const data = try self.stack.pop();
+        defer self.allocator.free(data);
+
+        var hash1: [Sha256.digest_length]u8 = undefined;
+        Sha256.hash(data, &hash1, .{});
+
+        var hash2: [Sha256.digest_length]u8 = undefined;
+        Sha256.hash(&hash1, &hash2, .{});
```
Comment (tdelabro) [inline src/script/engine.zig:499]:
```zig
    var digest: [Sha256.digest_length]u8 = undefined;
    Sha256.hash(data, &digest, .{});
    Sha256.hash(&digest, &digest, .{});
```

I believe this does compile and work. Please check it

**Review by tdelabro — CHANGES_REQUESTED** [review]
Nice work. Ty

File: src/script/engine.zig (hunk 1 above)
Comment (KhairallahA) [reply src/script/engine.zig:499]: @tdelabro, it works! loved it

**Comment by KhairallahA** [comment]
@tdelabro, done!

**Review by tdelabro — APPROVED** [review]

**Pull request merged**